src/meltano/cli/utils.py

```diff
@@ -679,3 +680,39 @@ def invoke(self, ctx):
             ctx.obj["tracker"].add_contexts(CliContext.from_click_context(ctx))
             ctx.obj["tracker"].track_command_event(CliEvent.started)
         super().invoke(ctx)
+
+
+def traverse_nested_settings(
+    dict_value,
+    nested_settings: list[list[tuple | (set | int)]],
+    current_nested_setting=(),
+):
+    """Create an array of tuples for nested settings."""
+    for sub_setting, sub_value in dict_value.items():
+        if isinstance(sub_value, dict):
+            # If the value is a dictionary, continue traversing
+            traverse_nested_settings(
+                sub_value,
+                nested_settings,
+                current_nested_setting + (sub_setting,),
+            )
+        else:
+            # When you reach a leaf node (non-dictionary value),
+            # append the accumulated path to final_sub_settings
+            nested_settings.append([current_nested_setting + (sub_setting,), sub_value])
+
+
+def get_non_interactive_flow_setting_and_value(
+    current_setting,
+    current_value,
+):
+    """Update setting and value for non interactive stdin flow."""
+    current_setting += (current_value,)
+    input_data = click.get_text_stream("stdin").read()
+    try:
+        # Check if input requires parsing
+        value = json.loads(input_data)
+    except json.JSONDecodeError:
+        value = input_data
+
+    return current_setting, value
```

Comment:
Yes there is still as issue with `meltano config meltano set cli '{"log_level": "debug"}'` it sets cli.log_level = '{"log_level": "debug"}' 
Probably we should fix this such that we dont have to write custom function for parsing.I'll fix it

Reply:
Sorry I Dint get it @edgarrmondragon

Reply:
@XshubhamX Thanks for raising that!

FWIW we used to try JSON-deserializing input before but removed it cause it was a bit problematic (see https://github.com/meltano/meltano/pull/7017). So I don't want to increase the scope of this PR with that.

Reply:
@edgarrmondragon  should we retain the function that traverses nested settings in this PR or should we just accept that we will only receive text information in files rather than JSON?

Reply:
@edgarrmondragon Then we'd have to specify that files should only contain value of a specific setting and not nested setting, That we can mention in docs.

Reply:
@edgarrmondragon can you confirm here?

Reply:
@XshubhamX Yeah, a quick mention in the docs that the setting has to be of `object` or `array` for it to be deserialized works for me.